### ISSUE TITLE
Improve godocs

### ISSUE THREAD

**Issue by codefromthecrypt** [tetratelabs/wazero#426]
I think our package docs could be improved. Also code snippets in our godocs aren't tested. we should change to normal examples to ensure they don't drift Ex. https://pkg.go.dev/github.com/tetratelabs/wazero#StartWASICommand

If someone has a good example to compare to as what "good godoc" is, happy to have references!

See https://github.com/tetratelabs/wazero/issues/425

**Comment by codefromthecrypt** [comment]
In most cases we can move code snippets embedded in godocs to Example tests like this https://github.com/golang/go/blob/master/src/os/exec/example_test.go

help wanted if someone has time, otherwise in a week or so I will do it.

**Comment by codefromthecrypt** [comment]
fyi due to this ancient issue in go, there's limited ability to improve docs on interfaces, as they neither break down methods, nor embed examples in them. smell of a hackweek https://github.com/golang/go/issues/5860

**Comment by codefromthecrypt** [comment]
a link to godocs is on the new website under API https://wazero.io/

**Comment by codefromthecrypt** [comment]
closing as I think our godocs are pretty good right now

**Comment by codefromthecrypt** [comment]
the formatting of godoc like below doesn't linefeed either. If someone complains, we should do something different (such as redo docs so that it isn't a problem), look for an issue there is in Go about formatting and cite it, or look to see a better formatted alternative to https://pkg.go.dev

```
// Note: All WASI functions return a single Errno result, ErrnoSuccess on success.
// Note: Closing the wazero.Runtime closes this instance of WASI as well.
```

Note: any godoc improvements should apply to all our public packages: wazero, wasi, assemblyscript, sys, api, and experimental. If we decide to do anything, we should do it to all the public packages, not just one of them.

**Comment by codefromthecrypt** [comment]
I looked into it and while there are many ways to re-render godoc into
markdown, there are limited options on what to do in godoc itself. There
are tricks implicitly used where preformatting (tab indent) is used to
avoid having to add extra lines everywhere. This is done in gob, for
example. I raised #600 to fix all the files that are public

See https://go.dev/src/encoding/gob/doc.go
See https://github.com/fluhus/godoc-tricks/blob/master/doc.go
